net/portmapper/portmapper.go

```diff
@@ -409,10 +414,18 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPor
 		}
 	}
 
-	// And ask for a mapping.
-	pmpReqMapping := buildPMPRequestMappingPacket(localPort, prevPort, pmpMapLifetimeSec)
-	if _, err := uc.WriteTo(pmpReqMapping, pmpAddru); err != nil {
-		return netaddr.IPPort{}, err
+	if haveRecentPMP {
+		// And ask for a mapping.
+		pmpReqMapping := buildPMPRequestMappingPacket(localPort, prevPort, pmpMapLifetimeSec)
+		if _, err := uc.WriteTo(pmpReqMapping, pmpAddru); err != nil {
+			return netaddr.IPPort{}, err
+		}
+	} else {
+		// Only do PCP mapping in the case when PMP did not appear to be available recently.
```

Comment:
so we send PCP unconditionally all the time even if we've never heard about its availability from an earlier probe?

I guess that's what we did for PMP previously?

Seems a little less than ideal. Maybe we should just bail out earlier if we haven't heard from anything.

Reply:
oops that's not what I intended, let me rewrite it